components/dashboards-web-component/src/gadgets-generation-wizard/components/chartPropertyGenerators/main/SearchBar.jsx

```diff
@@ -0,0 +1,74 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+import React, { Component } from 'react';
+import SwitchProperty from '../../inputTypes/SwitchProperty';
+import StreamProperty from '../../inputTypes/StreamProperty';
+
+/**
+ * Represents a search bar
+ */
+class SearchBar extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            configuration: props.configuration,
+            expandAdvanced: false,
+        };
+    }
+
+    /**
+     * Assigns value for the search bar's property, that has the given key
+     * @param key
+     * @param value
+     */
+    handleChartPropertyChange(key, value) {
+        const state = this.state;
+        state.configuration.charts[0][key] = value;
+        this.setState(state);
+        this.props.onConfigurationChange(state.configuration);
+    }
+
+    render() {
+        return (
+            <div>
+                <StreamProperty
+                    id="column"
+                    value={this.state.configuration.charts[0].column}
+                    fieldName="Column name*"
```

Comment:
The required fields are validated at the point of clicking 'Create' button. This is how it is currently implemented for all chart types in generation wizard.

Reply:
In other fields there are other validations performed against the input other than checking whether the input is null or " " 
Also that type of validation will return only a generic response to the user such as "required fields missing" but will not highlight inline which field is missing. 
Anyway for the moment +1 to continue with this approach as it requires considerable refactoring in some other components as well.

Reply:
The search bar widget is intended to list all distinct values of a user selected column. Since the column value will be selected from a select field which is populated with the names of the columns retrieved, the only validation needed is null or empty check.
+1 for inline validation as it will be more informative and user friendly.